addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -456,6 +459,7 @@ export async function loadImageInfo(img, rpc, attachmentSrc = "", resField = "")
     const isLocal = original && new URL(original.image_src, window.location.origin).origin === window.location.origin
         && !/\/web\/image\/\d+-redirect\//.test(original.image_src);
     if (isLocal && original.image_src) {
+        odooEditor.observerUnactive("loadImageInfo");
```

Comment:
Should not that be a fix in stable? (real question)

Reply:
We could add it in stable in order to be more accurate but it will not fix a flow. Indeed:
- If the image is not an embedded field (it is inside an `arch` for exemple -> image on the website main page): the `loadImageInfo` operation is done at the drop of the image on the page (through `_initializeImage`). At the save of the `arch` field, the `originalSrc` attribute is saved inside the field so the `loadImageInfo` process will not be done the next time the user edit the page and click on the image. In this case, `.o_dirty` is added because there is a new image on the page.
- If the image is linked to an embedded field (`image_1920` of `product.template` for example), the `loadImageInfo` process will only be done if the user upload a new image (and so `.o_dirty` has to be added). Indeed, in this case, the `/web_editor/get_image_info` rpc will return a correct `original` leading to the complete `loadImageInfo`process flow. If the image is not a new uploaded image, the rpc will not be able to return the correct `original` (Note that it is one of the goal of this task to adapt the `/web_editor/get_image_info`  rpc so that it returns the correct attachment linked to a field of a record).

-> In those two cases, the `loadImageInfo` process is done only if the image is new so `.o_dirty` has to be added anyway.
However, this fix should be added when the fix of the second commit of this task will be merged in stable as a the `loadImageInfo` process will be done for images that are not necessarily new on the page.

Reply:
I see, thanks for clarifying :+1: Although, feels strange in the end to actually prevent the o_dirty class to be added for a DOM change we would actually like to be saved in the DOM (to not go through `get_image_info` again) but I understand. We'll get read of those dataset values at some point anyways, we'll re-debate at that point. For now this is good like this, for master :+1: Maybe just add this clarification in the commit message itself? :+1: